scripts/tasks.py

```diff
@@ -509,3 +511,118 @@ def timetool_correct(config):
+    logger.info('Done!')
+    
+def bo_init_samples_configs(config):
+    from btx.diagnostics.bayesian_optimization import BayesianOptimization
+    setup = config.setup
+    task = config.bayesian_opt
+    """ Generates the config files that will be used to generate the initial samples for the Bayesian optimization. """
+    taskdir = os.path.join(setup.root_dir, "bayesian_opt")
+    os.makedirs(taskdir, exist_ok=True)
+    os.makedirs(os.path.join(taskdir, task.tag), exist_ok=True)
+
+    if hasattr(task, 'n_samples_init'):
+        n_samples_init = task.n_samples_init
+
+        # Get the parameters to overwrite
+        params_ranges_keys = [key for key in config if key.startswith("range_")]
+        params_names = [key.replace("range_", "") for key in params_ranges_keys]
+
+        # Generate the parameter samples
+        param_samples = BayesianOptimization.random_param_samples(config)
+
+        # Get the first task of the loop sequence
+        task_to_optimize = config.get(task.task_to_optimize)
+
+        # Get the task generating the scores
+        score_task = config.get(task.score_task)
+
+        # Generate the subdir
+        subdir_name = setup.exp + "_init_samples_configs"
+        subdir_path = os.path.join(setup.root_dir, "bayesian_opt", task.tag, subdir_name)
+        os.makedirs(subdir_path)
+
+        # Generate the config files
+        logger.info(f'Generating {n_samples_init} config files.')
+        for i in range(n_samples_init):
+            config_temp = config.copy()
+            # Overwrite the parameters
+            for j in range(len(params_names)):
+                config_temp[task_to_optimize][params_names[j]] = param_samples[i,j]
+
+            # Overwrite the score task tag to change the name of the score output file
+            config_temp[score_task]["tag"] = config_temp[score_task]["tag"] + f"_sample_{i+1}"
+
+            # Write the config file
+            config_file_name = f"{setup.exp}_sample_{i+1}.yaml"
+            config_file_path = os.path.join(subdir_path, config_file_name)
+
+            with open(config_file_path, 'w') as file:
+                yaml.dump(config_temp, file)
+            
+        logger.info('Done!')
+    else:
+        logger.info('The number of config files to generate was not defined!')
+
+def bo_aggregate_init_samples(config):
+    setup = config.setup
+    task = config.bayesian_opt
+    """ Aggregates the scores and parameters of the initial samples of the Bayesian optimization. """
+    n_samples_init = task.n_samples_init
+    # Get the names of the parameters
+    params_ranges_keys = [key for key in config if key.startswith("range_")]
+    params_names = [key.replace("range_", "") for key in params_ranges_keys]
+    n_params = len(params_names)
+    # Get the first task of the loop sequence
+    task_to_optimize = config.get(task.task_to_optimize)
+    # Get the task generating the scores
+    score_task = config.get(task.score_task)
```

Comment:
Similarly to within `BayesianOptimization` - could a single attribute access method be used for all the parameters?

Reply:
Done.